src-test/org/etools/j1939_84/controllers/part01/Part01Step08ControllerTest.java

```diff
@@ -325,13 +352,49 @@ public void testNoSpnNPacketsMatch() {
         assertEquals("", listener.getResults());
     }
 
+    /**
+     * Test packet with expected SPNs returned for global or destination specific
+     * DM20 request - spark engine
+     */
     @Test
+    @TestDoc(value = @TestItem(verifies = "6.1.8.2.a", description = "A.4 - Spark Ignition Engine Minimum SPs Verified: Expected SPNs"))
     public void testSparkIgnition() {
 
         List<Integer> SPNs = List.of(3054, 3058, 3306, 3053, 3050, 3051, 3055, 3056, 3057);
 
-        DM20MonitorPerformanceRatioPacket dm20 = createDM20(SPNs);
+        DM20MonitorPerformanceRatioPacket dm20 = createDM20(0, SPNs);
+
+        when(diagnosticMessageModule.requestDM20(any())).thenReturn(RequestResult.of(dm20));
+
+        dataRepository.putObdModule(new OBDModuleInformation(0));
+
+
+        VehicleInformation vehicleInformation = new VehicleInformation();
+        vehicleInformation.setFuelType(FuelType.BI_CNG);
+        dataRepository.setVehicleInformation(vehicleInformation);
+
+        runTest();
+
+        verify(diagnosticMessageModule).requestDM20(any());
+
+        assertEquals("", listener.getMessages());
+        assertEquals("", listener.getResults());
+    }
 
+    /**
+     * Test packet with expected SPNs returned for global or destination specific
+     * DM20 request - spark engine
+     */
+    @Test
+    @TestDoc(value = @TestItem(verifies = "6.1.8.2.a", description = "When a numerator and denominator are provided as FFFFh and FFFFh, the monitor identified in the label SPN shall be considered to be unsupported"))
+    public void testAllNumeratorAndDenominatorAllFs() {
+        int moduleAddress = 0;
+        int spn = 524287;
```

Comment:
The test should have detected that SPN 524287 wasn’t supported. I think this SPN should be switched to one of the required SPNs.  I’d write the test first with good values and no failures, then change the values to 0xFFFF to produce the failure.

Reply:
[Fixed](https://github.com/SolidDesignNet/j1939-84/pull/15/commits/ee27fabc261386b5077c7010f9885f660d890a86)